File: Services/Inquiry/Tameenk.Services.InquiryGateway/Models/InquiryRequestModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Tameenk.Services.InquiryGateway.Models.ExceptionHandlerModel;

namespace Tameenk.Services.InquiryGateway.Models
{
    /// <summary>
    /// The inquery request parameter.
    /// </summary>
    [JsonObject("inqueryRequest")]
    public class InquiryRequestModel
    {
        /// <summary>
        /// The city code.
        /// </summary>
        [JsonProperty("cityCode")]
        [Required]
        public long CityCode { get; set; }

        /// <summary>
        /// The policy effective date.
        /// </summary>
        [JsonProperty("policyEffectiveDate")]
        [Required]
        public DateTime PolicyEffectiveDate { get; set; }


        /// <summary>
        /// Is the vehicle user commercially.
        /// </summary>
        [JsonProperty("isVehicleUsedCommercially")]
        public bool IsVehicleUsedCommercially { get; set; }


        /// <summary>
        /// Old owner national identification number.
        /// </summary>
        [JsonProperty("oldOwnerNin")]
        public long? OldOwnerNin { get; set; }
        
        /// <summary>
        /// Is customer have special need.
        /// </summary>
        [JsonProperty("isCustomerSpecialNeed")]
        public bool? IsCustomerSpecialNeed { get; set; }

        /// <summary>
        /// The addtional driver(s).
        /// </summary>
        [JsonProperty("drivers")]
        public List<DriverModel> Drivers { get; set; }

        /// <summary>
        /// The insured person information.
        /// </summary>
        [JsonProperty("insured")]
        public InsuredModel Insured { get; set; }

        /// <summary>
        /// The vheicle.
        /// </summary>
        [JsonProperty("vehicle")]
        public VehicleModel Vehicle { get; set; }

        /// <summary>
        /// User captcha input.
        /// </summary>
        [JsonProperty("captchaInput")]
        public string CaptchaInput { get; set; }

        /// <summary>
        /// The captcha token.
        /// </summary>
        [JsonProperty("captchaToken")]
        public string CaptchaToken { get; set; }

        [JsonProperty("isEditRequest")]
        public bool IsEditRequest { get; set; }
        [NotMapped]
        [JsonProperty("inquiryOutputModel")]
        public InquiryOutputModel InquiryOutputModel { get; set; }

    }
}

File: Tameenk/src/app/home/models/inquiry-request.model.ts
import { Driver } from "./driver.model";
import { Vehicle } from "./vehicle.model";
import { Insured } from "./insured.model";
import { Captcha } from "./captcha.model";
export interface InquiryRequest {
  /**
   * @property {number} cityCode - The City Driving Code.
   */
  cityCode: number;
  /**
   * @property {string} policyEffectiveDate - The Policy Effective Date.
   */
  policyEffectiveDate: string;

  // isVehicleUsedCommercially: boolean;
  /**
   * @property {boolean} isCustomerCurrentOwner - Is customer Current Owner.
   */
  isCustomerCurrentOwner: boolean;
  /**
   * @property {number} oldOwnerNin - The Vehicle Old Owner National Id.
   */
  oldOwnerNin: number;

  // isCustomerSpecialNeed: boolean;
  /**
   * @property {Driver[]} drivers - drivers model
   */
  drivers: Driver[];
  /**
   * @property {Insured} insured - insured model
   */
  insured: Insured;
  /**
   * @property {Vehicle} vehicle - vehicle model
   */
  vehicle: Vehicle;
    /**
   * @property {Captcha} captcha - The captcha Model.
   */
  captcha: Captcha;
  isMainDriverExist: boolean;
  isVehicleExist: boolean;
}
